sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -289,9 +289,11 @@ export function appendQueryParams(
     if (typeof value === "string") {
       searchPieces.push(`${name}=${value}`);
     } else {
```

Comment:
can avoid a level of nesting here by changing this to else/if

```suggestion
    } else if(Array.isArray(value)) {
```